DigitalPuzzle/DigitalPuzzle.ino

```diff
@@ -355,61 +244,284 @@ void ReadSerial()
+  readers[0].Inputs[1] = &wires[1];
+  readers[0].Output    = &wires[2];
 
+  //READER 1
+  readers[1] = portReader(portReader::NO_GATE);
 
+  wires[3] = LedStrip(H, 7, 6);
+  wires[4] = LedStrip(L, 5, 4);
+  wires[5] = LedStrip(Z, 17, 16);
+  readers[1].Inputs[0] = &wires[3];
+  readers[1].Inputs[1] = &wires[4];
+  readers[1].Output    = &wires[5];
 
-void setup()
-{
-  Serial.begin(9600); 
-  // EEPROM.put(0,164);
-  // EEPROM.put(1,175); 
-  // EEPROM.put(2,3); 
-  // EEPROM.put(3,11); 
-  // EEPROM.put(10,213); 
-  // EEPROM.put(11,61); 
-  // EEPROM.put(12,4); 
-  // EEPROM.put(12,4); 
-  // EEPROM.put(13,50);
-  for(int i; i<2; i++)
-  {
-    pinMode(RedPins[i], OUTPUT);
-    pinMode(GreenPins[i], OUTPUT);
-  
-    digitalWrite(RedPins[i], LOW);
-    digitalWrite(GreenPins[i], LOW);
+  //READER 2
+  readers[2] = portReader(portReader::NO_GATE);
+
+  wires[6] = LedStrip(H, 1, 0);
+  wires[7] = LedStrip(NC, -1, -1);
+  wires[8] = LedStrip(Z, 13, 12);
+  readers[2].Inputs[0] = &wires[6];
+  readers[2].Inputs[1] = &wires[7];
+  readers[2].Output    = &wires[8];
+
+  //READER 3
+  readers[3] = portReader(portReader::NO_GATE);
+
+  wires[9] = LedStrip(NC, -1, -1);
+  wires[10] = LedStrip(Z, 23, 22);
+  readers[3].Inputs[0] = readers[0].Output;
+  readers[3].Inputs[1] = &wires[9];
+  readers[3].Output    = &wires[10];
+
+  //READER 4
+  readers[4] = portReader(portReader::NO_GATE);
+
+  wires[11] = LedStrip(H, 3, 2);
+  wires[12] = LedStrip(Z, 21, 20);
+  readers[4].Inputs[0] = readers[1].Output;
+  readers[4].Inputs[1] = &wires[11];
+  readers[4].Output    = &wires[12];
+
+  //READER 5
+  readers[5] = portReader(portReader::NO_GATE);
+
+  wires[13] = LedStrip(Z, 27, 26);
+  readers[5].Inputs[0] = readers[3].Output;
+  readers[5].Inputs[1] = readers[1].Output;
+  readers[5].Output    = &wires[13];
+
+  //READER 6
+  readers[6] = portReader(portReader::NO_GATE);
+
+  wires[14] = LedStrip(Z, 25, 24);
+  readers[6].Inputs[0] = readers[4].Output;
+  readers[6].Inputs[1] = readers[2].Output;
+  readers[6].Output    = &wires[14];
+
+  //READER 7
+  readers[7] = portReader(portReader::NO_GATE);
+
+  wires[15] = LedStrip(Z, 29, 28);
+  readers[7].Inputs[0] = readers[5].Output;
+  readers[7].Inputs[1] = readers[6].Output;
+  readers[7].Output    = &wires[15];
+
+  for (int i = 0; i < AMOUNT_OF_READERS; i++) {
+	  readers[i].CalculateOutput();
   }
+
+
+  for (int i = 0; i < 32; i++) {
+	  sr.set(i, HIGH);
+	  delay(10);
+  }
+
+  for (int i = 0; i < 32; i++) {
+	  sr.set(i, LOW);
+	  delay(10);
+  }
+
+  RandomizeStartValues();
+  Serial.print(StartValues[0]);
+  Serial.print(StartValues[1]);
+  Serial.print(StartValues[2]);
+  Serial.print(StartValues[3]);
+  Serial.print(StartValues[4]);
+  Serial.println(StartValues[5]);
   Serial.println("Setup Done!");
 }
 
 void loop()
 {
-	ReadSerial();
-    ApplyResults();
-    //CompareNFC(ReadRFID(0));
-    //CompareNFC(ReadRFID(1));
-}
\ No newline at end of file
+  ReadSerial();
```

Comment:
Deze functie moet bij nader inzien ook iets discriptiever. Het is vanuit hier niet duidelijk wat er precies gebeurt in ReadSerial

Reply:
ReadSerialPort misschien beter. Eigenlijk is die ReadSerial vooral voor het debuggen zonder de control unit/kast. In de EscapeRoom krijg ik toch commando's van de CU right?